File: Semestrul 5/Proiectarea-Detaliata-a-Sistemelor-Software/Laborator/Daria/Laborator_10/decorator/Mushroom.java
package Laborator.Daria.Laborator_10.decorator;

public class Mushroom extends PizzaDecorator {
    public Mushroom(Pizza pizza) {
        super(pizza);
    }

    @Override
    public String getDescription() {
        return super.getDescription() + ", Mushroom";
    }

    @Override
    public double getCost() {
        return super.getCost() + 4.0;
    }
}

File: Semestrul 5/Proiectarea-Detaliata-a-Sistemelor-Software/Laborator/Paul/design-patterns/decorator/src/main/kotlin/pizza/ingredients/Mushroom.kt
package org.paul.pizza.ingredients

import org.paul.pizza.Pizza
import org.paul.pizza.PizzaDecorator

class Mushroom(pizza: Pizza) : PizzaDecorator(pizza) {
    override fun getCost(): Double {
        return super.getCost() + COST
    }

    override fun getDescription(): String {
        return super.getDescription() + ", Mushroom"
    }

    companion object {
        private const val COST = 4.0
    }
}
